connect/src/main/java/com/stripe/android/connect/webview/StripeConnectWebViewClient.kt

```diff
@@ -0,0 +1,167 @@
+                $ANDROID_JS_INTERFACE.pageDidLoad = (message) => {
+                    $ANDROID_JS_INTERNAL_INTERFACE.pageDidLoad(JSON.stringify(message));
+                };
+                $ANDROID_JS_INTERFACE.fetchClientSecret = () => {
+                    return new Promise((resolve, reject) => {
+                        try {
+                            let clientSecret = $ANDROID_JS_INTERNAL_INTERFACE.fetchClientSecret()
+                            resolve(clientSecret);
+                        } catch (e) {
+                            reject(e);
+                        }
+                    });
+                };
+                $ANDROID_JS_INTERFACE.onSetterFunctionCalled = (message) => {
+                    $ANDROID_JS_INTERNAL_INTERFACE.onSetterFunctionCalled(JSON.stringify(message));
+                };
+                $ANDROID_JS_INTERFACE.openSecureWebView = (message) => {
+                    $ANDROID_JS_INTERNAL_INTERFACE.openSecureWebView(JSON.stringify(message));
+                };
+            """.trimIndent(),
+            null
+        )
+    }
+
+    private inner class StripeJsInterface {
+        @JavascriptInterface
+        fun debug(message: String) {
+            logger.debug("Debug log from JS: $message")
+        }
+
+        @JavascriptInterface
+        fun fetchInitParams() {
+            logger.debug("InitParams fetched")
+        }
+
+        @JavascriptInterface
+        fun fetchInitComponentProps() {
+            logger.debug("InitComponentProps fetched")
+        }
+    }
+
+    private inner class StripeJsInterfaceInternal {
+        @JavascriptInterface
+        fun log(message: String) {
+            logger.debug("Log from JS: $message")
+        }
+
+        @JavascriptInterface
+        fun onSetterFunctionCalled(message: String) {
+            val setterMessage = jsonSerializer.decodeFromString<SetterMessage>(message)
+            logger.debug("Setter function called: $setterMessage")
+        }
+
+        @JavascriptInterface
+        fun openSecureWebView(message: String) {
+            val secureWebViewData = jsonSerializer.decodeFromString<SecureWebViewMessage>(message)
+            logger.debug("Open secure web view with data: $secureWebViewData")
+        }
+
+        @JavascriptInterface
+        fun pageDidLoad(message: String) {
+            val pageLoadMessage = jsonSerializer.decodeFromString<PageLoadMessage>(message)
+            logger.debug("Page did load: $pageLoadMessage")
+        }
+
+        @JavascriptInterface
+        fun accountSessionClaimed(message: String) {
+            val accountSessionClaimedMessage = jsonSerializer.decodeFromString<AccountSessionClaimedMessage>(message)
+            logger.debug("Account session claimed: $accountSessionClaimedMessage")
+        }
+
+        @JavascriptInterface
+        fun fetchClientSecret(): String {
+            return runBlocking {
+                EmbeddedComponentManager.getInstance().fetchClientSecret() ?: ""
```

Comment:
should this error if null?
```suggestion
                checkNotNull(EmbeddedComponentManager.getInstance().fetchClientSecret())
```

Reply:
Yes good catch. Throwing here will cause the JS code I wrote to reject the promise, which should correctly propagate it to the Connect SDK. Thanks!